lib/puppet/util/windows/security.rb

```diff
@@ -372,7 +372,10 @@ def set_mode(mode, path, protected = true)
     dacl.allow(well_known_nobody_sid, nobody_allow)
 
     # TODO: system should be first?
-    dacl.allow(well_known_system_sid, system_allow)
+    flags = !isdir ? 0 :
+      Puppet::Util::Windows::AccessControlEntry::CONTAINER_INHERIT_ACE |
+      Puppet::Util::Windows::AccessControlEntry::OBJECT_INHERIT_ACE
+    dacl.allow(well_known_system_sid, system_allow, flags)
```

Comment:
Are there places where we don't want to allow this inheritance? Or do we believe it's fine to always allow `System` access. If so... :+1:.